src/tests/electron/tests/automated-checks-view.test.ts

```diff
@@ -67,7 +72,44 @@ describe('AutomatedChecksView', () => {
         await cardsView.assertExpandedRuleGroup(3, 'TouchSizeWcag', 1);
     });
 
-    it('should pass accessibility rvalidation in all contrast modes', async () => {
+    it('supports expanding and collapsing rule groups with results_v2', async () => {
+        app.setFeatureFlag(UnifiedFeatureFlags.atfaResults, true);
+        await openResultsAndCardsViews();
+
+        await cardsView.waitForHighlightBoxCount(3);
+        expect(await cardsView.queryRuleGroupContents()).toHaveLength(0);
+
+        await cardsView.toggleRuleGroupAtPosition(1);
+        await cardsView.assertExpandedRuleGroup(1, 'EditTextValue', 1);
+
+        await cardsView.toggleRuleGroupAtPosition(2);
+        await cardsView.assertExpandedRuleGroup(2, 'TouchSizeWcag', 1);
+
+        await cardsView.toggleRuleGroupAtPosition(3);
+        await cardsView.assertExpandedRuleGroup(3, 'ImageViewName', 1);
+
+        await cardsView.waitForHighlightBoxCount(3);
+        expect(await cardsView.queryRuleGroupContents()).toHaveLength(3);
+
+        await cardsView.toggleRuleGroupAtPosition(1);
+        await cardsView.assertCollapsedRuleGroup(1, 'EditTextValue');
+
+        await cardsView.toggleRuleGroupAtPosition(2);
+        await cardsView.assertCollapsedRuleGroup(2, 'TouchSizeWcag');
+
+        await cardsView.waitForHighlightBoxCount(1);
+        expect(await cardsView.queryRuleGroupContents()).toHaveLength(1);
+        await cardsView.assertExpandedRuleGroup(3, 'ImageViewName', 1);
+    });
+
+    it('should pass accessibility validation in all contrast modes', async () => {
+        await openResultsAndCardsViews();
         await scanForAccessibilityIssuesInAllModes(app);
     });
+
+    async function openResultsAndCardsViews(): Promise<void> {
```

Comment:
Potential refactor (maybe a separate PR): If this returned `Promise<ResultsViewController>`, then `resultsView` could be scoped inside each test method.